### ISSUE TITLE
Improve googleability by using real error messages in documentation

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#2393]
For example, http://mypy.readthedocs.io/en/latest/common_issues.html could have the actual mypy error messages for examples to make it easier to Google for common errors.

**Comment by nathanielksmith** [comment]
I think this is something I can take on. I'll run through the examples and add actual error messages where appropriate.

**Comment by gvanrossum** [comment]
There's more than that particular chapter of the docs right?

**Comment by nathanielksmith** [comment]
Yup, and I'm reading the docs through now to look for more stuff. I figured I'd eagerly PR though to make sure my approach is ok.

I'll update my PR description though to be more clear that it's incremental.
